src/Platform/Microsoft.Testing.Platform/Builder/TestApplication.cs

```diff
@@ -78,6 +78,11 @@ public static async Task<ITestApplicationBuilder> CreateBuilderAsync(string[] ar
         // First step is to parse the command line from where we get the second input layer.
         // The first one should be the env vars handled autonomously by extensions and part of the test platform.
         CommandLineParseResult parseResult = CommandLineParser.Parse(args, systemEnvironment);
+        if (parseResult.IsOptionSet(PlatformCommandLineProvider.DebugAttachOptionKey))
```

Comment:
On a first glance, I would interpret `--debug-attach` as "it attaches the debugger" (i.e, it does `Debugger.Launch`) not that "it waits for the debugger to be attached"

Reply:
`--debug-wait-attach`?

Reply:
Works for me